BUILD.md

```diff
@@ -557,3 +557,19 @@ Build and generate report:
 meson setup build
 ninja -C build scan-build
 ```
+
+### Installation
+
+You can specify directory where files will be installed using `DESTDIR`
+environment variable, and configure prefix and directory paths using `setup`
+parameters:
```

Comment:
Meson expands the user base to macOS and Windows users (which is awesome), but for these users with a Linux or auto-tools background, `configure prefix` might be unknown.  

I suppose DESTDIR also has its heritage in auto-tools, but on the flip side Linux users will have:

`"./configure --prefix=... && make && sudo make install` burned into people's minds, as opposed to: `... sudo DESTDIR=../.. make install`. 

I wonder if Meson has a nice blurb about controlling the install directory? 

My guess is that almost all users doing stuff on the command line will want `dosbox` installed to an existing PATH either off there $HOME directory or in their OSes equivalent of `/usr/local`. 

So we might want to provide a handful of examples that achieves that. Otherwise pure newbies might make a mess, and we'll be answering questions of "where did my dosbox binary get installed"? 

The more we can recommend, the more uniform we can support everyone's environment.  I'm not sure if meson has those examples for the big-three OSes?

Reply:
OK, I think we need to be clear - this documentation is not for end users, it's for developers and packagers - consequently:

- I don't care if Windows users understand this or not - it's not for them
- macOS users should never use this documentation - it's for `brew` packagers and the like
- Linux users should never use this documentation - it's directed mostly for packagers building for various Linux-based OSes/solutions

Consequently, I think I am going to remove this from BUILD.md file.

Reply:
For this **Installation** section I'm envisioning we start easy and give progressively more detail and options.

We recommend installating your compiled binary into your account's directory so you can easily access it from your desktop application list or on the command-line. 

 - **Windows users**:
     - `meson setup --prefix="$HOMEPATH" build` followed by `meson install -C build`
     - or maybe [one of these.](https://www.itprotoday.com/cloud-computing/what-environment-variables-are-available-windows)
     - But we certainly don't want pollution like meson's [Windows default](https://github.com/mesonbuild/meson/issues/1905).
 - **MacOS users**:
    - `meson setup --prefix=/usr/local build && ninja -C build install` (as this is the[ most common path for manually-built user executables](https://apple.stackexchange.com/questions/80902/where-should-shell-tools-be-installed)).
 - Linux users:
    - `meson setup --prefix="$HOME/.local" build && ninja -C build install`
 
After configuration compiler and install with:

``` shell
cd build

Reply:
> Consequently, I think I am going to remove this from `BUILD.md` file.

It makes sense that we separate the documentation targeting packager maintainers whose job involves creating OS packages for end-user consumption, versus `BUILD.md` which targets those willing to make the jump (or power-users) wanting to compile Staging in as few steps as possible, which is the goal of `BUILD.md`.

I believe we can do that even more elegantly with Meson:
 1) Install Python (essentially Windows-only)
 2) Install or update PIP (a command or two)
 3) Install latest-meson (a command)
 4) Install GIT (a common or a couple steps)
 5) Clone Staging
 6) Configure, build, install w/ Meson (one or two steps)

Reply:
I would consider PACKAGING.md outsite the scope of this PR.

But if you can add simply recommended install paths (for the end user, like in my example) that would be very nice! I'll be ready to try those steps :-)

Reply:
As for paths:

- We don't want Windows users to attempt `install` via Meson, ever. It only makes sense for people who use MSYS2 or Cygwin and know what they are doing. It's explained very well in the discussion under Meson bug report you mentioned:

  > bindir, libdir, etc, are concepts that don't translate to native Windows applications.

  In other words: `install` target on Windows makes about as much sense as storing Linux programs in "~/Program Files (x86)" directory… Meson provides this option only for people turning their Windows OS into Unix-like OS.

  What we should do is rather provide Windows-specific targets for creation of a zip file and installer file - but it's out of scope of this PR.

- For macOS users - we don't need to document anything right now, `/usr/local` is the default prefix already (using this prefix is also customary on Linux btw - `/usr/local` hierarchy is for software that's not managed by OS's primary package manager). I will probably move our App Bundle creation script so it can be invoked via meson (something like `meson setup build && ninja -C build app-bundle`; also out of scope of this PR).

I don't know how to handle this documentation without doing a disservice to the users. In a normal situation, users should never attempt to install software via `make install` or `ninja install` (with or without sudo). I removed the whole documentation section for now, I need to think how to communicate this while keeping our documentation lean and readable.